### ISSUE TITLE
Pressing Super key on virtual keyboard making the entire keyboard freeze with super stuck  on Hyprland Compositor

### ISSUE THREAD

**Issue by openwick** [jjsullivan5196/wvkbd#81]
**Tldr:**  Using only super key makes keyboard freeze with super key stuck  on hyprland  compositor and I uploaded a small video of the bug 

[wvkbd.webm](https://github.com/user-attachments/assets/8a6f31e3-a439-47f4-be65-a45e3aee6b40)

**Long post:** First of all thank you for making this. I am trying to find a on screen virtual keyboard for [Hyprland](https://github.com/hyprwm/Hyprland) compositor  none of virtual keyboard doesn't work except this one. However my only gripe is that I cant use super key for some reason as pressing it makes the super key stuck and freeze the entire keyboard and I can only undo it by pressing ctrl+alt+f2 or f3 and logging out using tty. 

I don't know if it a hyprland bug or keyboard bug but based on seeing everything works with the keyboard except super key I am guessing it the keyboard issue hence I am filing the issue here? Please let me know if I made any mistake and how to solve it.

PS: Also is there a way to auto hide and make the keyboard pop up like on android or ios or like maliit on KDE  without staying there static all the time? Also is there a way to adjust the size too cause that would be great.

**Comment by proycon** [comment]
This is a tricky situation: what's happening is that Super is configured as the main modifier key in your Hyprland config (as it is in mine). So as soon as you press Super in wvkbd, it's as if you're holding the super key on a physical keyboard and any subsequent click you do will be Super + click; which is by default bound to moving the window! This leads to the situation you're in that you can no longer click the super key because Hyprland will intercept it and wvkbd never gets the click. 

I wouldn't really say this is a bug in either wvkbd or Hyprland (or any other compositor for that matter), but it's an edge-case that will occur for keys that are the main modifier key in a compositor. I can't really think of an elegant way around this. Of course you can disable that particular keybinding (super + left mouse button) in Hyprland but you probably rely on it.

> PS: Also is there a way to auto hide and make the keyboard pop up like on android or ios or like maliit on KDE without staying there static all the time? 

Currently, no. See https://github.com/jjsullivan5196/wvkbd/issues/60

> Also is there a way to adjust the size too cause that would be great.

You can specify a size (height) on launch via a command line parameter.

**Comment by openwick** [comment]
Thanks for this. Currently there is also an [Issue](https://github.com/hyprwm/Hyprland/issues/7135) being discussed at hyprland too which provided a solution. I kept this is my binds conf and mostly it is working fine :)

```bash
#wvkbd
unbind=$mainMod,mouse:272
bindmn =$mainMod,mouse:272, movewindow
```
I think there are still issues with it but i think its works for now :)
